### PR TITLE
ARM7 OS_ splitting/porting

### PR DESCRIPTION
**Pull request by shinyquagsire23**
Direct ports from pokediamond w/o changes:
 - OS_mutex
 - OS_message
 - OS_context

Minor changes:
 - OS_thread

### CODE REVIEW

File: sub/include/function_target.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#ifndef GUARD_FUNCTION_TARGET_H
```
Comment (PikalaxALT) [inline sub/include/function_target.h:1]: Use the top level sdk include paths

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: include/nitro/OS_alarm_shared.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/*
```
Comment (PikalaxALT) [inline include/nitro/OS_alarm_shared.h:1]: move to `lib/nitro/include/os/common/alarm_shared.h`

File: sub/lib/include/OS_alarm.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#ifndef NITRO_SUB_OS_ALARM_H
```
Comment (PikalaxALT) [inline sub/lib/include/OS_alarm.h:1]: move to `lib/nitro/include/os/ARM7/alarm.h`

File: include/function_target.h [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#ifndef GUARD_FUNCTION_TARGET_H
```
Comment (PikalaxALT) [inline include/function_target.h:1]: This project will not make use of `function_target.h` in any capacity.

**Review by PikalaxALT — CHANGES_REQUESTED** [review]
The comments I left on the two NitroSDK headers are generally applicable to all NitroSDK headers. I despise the way the headers are organized in pokediamond, and will not permit the same structure in pokeheartgold.

**Comment by shinyquagsire23** [comment]
> The comments I left on the two NitroSDK headers are generally applicable to all NitroSDK headers. I despise the way the headers are organized in pokediamond, and will not permit the same structure in pokeheartgold.

ok yeah I kinda hated the organization as well, the comments def cleared up my confusion. Also kinda missed the entire `lib/` folder even existing so I'll refactor over to that then.

**Review by PikalaxALT — APPROVED** [review]

**Pull request merged**